lib/mkdirs/make-dir.js

```diff
@@ -0,0 +1,137 @@
+// Adapted from https://github.com/sindresorhus/make-dir
+// Copyright (c) Sindre Sorhus <sindresorhus@gmail.com> (sindresorhus.com)
+// Permission is hereby granted, free of charge, to any person obtaining a copy of this software and associated documentation files (the "Software"), to deal in the Software without restriction, including without limitation the rights to use, copy, modify, merge, publish, distribute, sublicense, and/or sell copies of the Software, and to permit persons to whom the Software is furnished to do so, subject to the following conditions:
+// The above copyright notice and this permission notice shall be included in all copies or substantial portions of the Software.
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+'use strict'
+const fs = require('../fs')
+const path = require('path')
+
+// Check for Node v10.12.0+
```

Comment:
I believe this doesn't work for 2 reasons!

1. If we use `process.version`, which has the value of something like `v12.5.0`, then the `nodeVersionMajor` will be NaN. If we use `process.versions.node`, then it should work because `process.versions.node` value is like `12.5.0` and doesn't have the `v` in it.
2. The check for `useNativeRecursiveOption` doesn't work! For instance, the result of the boolean check will be false for version `12.5.0` (which is greater than `10.12.0`). Although `nodeVersionMajor >= 10` but `nodeVersionMinor` is not `>= 12` and the result will be false while it should return true. You can take a loot at what I did in `util/stat.js: nodeSupportsBigInt()`. I wrote that when I was working on the `bigint` option issue. Just to tell you, that is definitely not the best implementation and I would become really happy if you come up with more beautiful solution :smiley:

Reply:
Yes, you're right on both counts. I actually wrote a module for this (that doesn't have these bugs); and I intended to suggest using it everywhere after this was merged: https://github.com/RyanZim/at-least-node